python/common/org/python/types/Float.java

```diff
@@ -423,9 +425,11 @@ public boolean __setattr_null(java.lang.String name, org.python.Object value) {
                     double result = (((((double) this.value) % other_val) + other_val) % other_val);
                     return new org.python.types.Float(result);
                 }
+            } else if (other instanceof org.python.types.Complex){
+                throw new org.python.exceptions.TypeError("can't mod complex numbers.");
             }
         } catch (org.python.exceptions.TypeError e) {
-            throw new org.python.exceptions.TypeError("unsupported operand type(s) for %: 'float' and '" + other.typeName() + "'");
+            throw e;
```

Comment:
If we're just rethrowing this exception, maybe we should simply not catch `TypeError` and let it raise.

Reply:
So should I remove try catch block ?